src/app/core/data/search.service.ts

```diff
@@ -18,79 +26,58 @@ export interface Option {
 })
 export class SearchService {
 
-  private nextUpdate = 0;
-  private readonly data = new BehaviorSubject<Option[]>([]);
+  private readonly index = new BehaviorSubject<Index | null>(null);
+  private readonly data = new BehaviorSubject<Record<number, Option[]>>({});
 
   constructor(
     private readonly http: HttpClient,
-  ) { }
-
-  private update() {
-    const now = Date.now();
-    if (this.nextUpdate < now) {
-      this.nextUpdate = now + 1000 * 60 * 10;
-      this.http.get<Record<string, Omit<Option, "name">>>(`${document.getElementsByTagName('base')[0].href}options.json`)
-        .subscribe(data => this.data.next(Object.entries(data).map(([name, data]) => ({ name, ...data }))));
-    }
+  ) {
+    from(__wbg_init(`${document.getElementsByTagName('base')[0].href}fixx_bg.wasm`))
+      .pipe(switchMap(() => this.http.get(`${document.getElementsByTagName('base')[0].href}index.ixx`, { responseType: 'arraybuffer' })))
+      .subscribe(data => this.index.next(Index.read(new Uint8Array(data))));
   }
 
-  public search(query: string): Observable<Option[]> {
-    this.update();
-
-    const search = query.toLowerCase().split('*');
-    if (search.length === 0) {
-      return of([]);
-    }
-
-    return this.data.pipe(map(options => {
-      const result = [];
+  public search(query: string | null | undefined): Observable<SearchedOption[]> {
+    return this.index.pipe(
+      map(index => {
+        return index ? (query && query.length > 0 ? index.search(query, 500).map(option => {
```

Comment:
We should move the 500 into a const, too.

Reply:
Yeah, I think it is currently referenced twice.